File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int currentSum=nums[0], maxSum=nums[0],k=nums.size();
        for(int i=1;i<k;i++){
            currentSum=max(nums[i],currentSum+nums[i]);
            maxSum=max(maxSum,currentSum);
        } 
        return maxSum;
    }
};

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currS=nums[0]
        maxS=nums[0]
        for i in range(1,len(nums)):
            currS=max(nums[i],nums[i]+currS)
            maxS=max(currS,maxS)

        return maxS
        